stdlib/public/core/Integers.swift.gyb

```diff
@@ -424,46 +1729,202 @@ public func > <T : BinaryInteger>(lhs: T, rhs: T) -> Bool {
 //===--- FixedWidthInteger ------------------------------------------------===//
 //===----------------------------------------------------------------------===//
 
+/// An indicator of whether an arithmetic operation overflowed.
+///
+/// Some arithmetic operations on fixed-width integers return an
+/// `ArithmeticOverflow` instance to indicate whether an overflow has
+/// occurred. For example, adding `UInt8.max` to itself results in a value that
+/// can't be represented by an `UInt8` instance without overflowing.
+///
+///     let x = UInt8.max
+///     // x == 255
+///     let (y, overflow) = x.addingWithOverflow(x)
+///     // y == 254
```

Comment:
We certainly do want to give such a guarantee.  Otherwise how would we build multiprecision addition?